### PR TITLE
Assert that MutableBox data is an IRNode

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97251
* #97294
* #97275

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Comment by ezyang** [comment]
This doesn't work! 😠

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def _check_tensorbox(node):
             (
                 DynamicScalar,
                 TensorBox,
+                MultiOutput,
```
Comment (ezyang) [inline torch/_inductor/ir.py:96]: This change is bad, because everything should be TensorBoxes

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -93,6 +93,8 @@ def _check_tensorbox(node):
             (
```
Comment (ezyang) [inline torch/_inductor/ir.py:93]: This function is actually about what valid RETURN types from Inductor are, this isn't just IR

File: torch/_inductor/ir.py (hunk 1 above)
Comment (wconstab) [reply torch/_inductor/ir.py:96]: hmm, now that you pointed out the above (return type) isn't it valid to return tuple(tensor) which is kind of what MultiOutput is?

**Pull request closed without merging**